### ISSUE TITLE
className with object containing toString() method isn't working as expected

### ISSUE THREAD

**Issue by resetko** [Semantic-Org/Semantic-UI-React#2599]
### Steps

1. Put object with `toString()` method to className prop for Button or any other component
2. Check out generated class name

Code example:
```js
import React from "react";
import { render } from "react-dom";
import { Button } from "semantic-ui-react";

class BEM {
  constructor(blockName) {
    this.block = blockName;
  }

  el(element) {
    return {
      mod: modf => `${this.block}__${element}--${modf}`,
      toString: () => `${this.block}__${element}`
    };
  }

  toString() {
    return `${this.block}`;
  }
}

const App = () => {
  const css = new BEM("block");
  return (
    <div className={css}>
      <button className={css.el("native")}>Native button</button>
      <br />
      <Button className={css.el("semantic")}>Semantic button</Button>
    </div>
  );
};

render(<App />, document.getElementById("app"));
```

### Expected Result
```html
<div id="app">
    <div class="block">
        <button class="block__native">Native button</button>
        <br>
        <button class="ui button mod block_semantic" role="button">Semantic button</button>
    </div>
</div>
```
### Actual Result
```html
<div id="app">
    <div class="block">
        <button class="block__native">Native button</button>
        <br>
        <button class="ui button mod toString" role="button">Semantic button</button>
    </div>
</div>
```
With warning for propType violation
### Version
0.78.3

### Testcase
https://codesandbox.io/s/j25zwlxmmw

**Comment by brianespinosa** [comment]
@resetko your `BEM` class is returning an object. If you take a look at the console in your codesandbox example, you are getting a PropType warning that className is expecting a string, not an object.

**Comment by resetko** [comment]
@brianespinosa Exactly. But this object is working fine with "native" react components, like `div`, or `button` and not working with SemanticUI. I'm concerning this as a bug, because you even have same prop name for your components as in React (`className`). If it's not a bug from your side - I would like to request such feature to make it work same as it's working for "native" react components.

**Comment by layershifter** [comment]
We heavy rely on [classnames](https://github.com/JedWatson/classnames) for `className` operations. As I see, the problem should be solved there, JedWatson/classnames#153.
